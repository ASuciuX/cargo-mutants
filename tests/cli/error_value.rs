// Copyright 2023 Martin Pool

//! Tests for error value mutations, from `--error-value` etc.

use std::env;

use predicates::prelude::*;

use super::{copy_of_testdata, run};

#[test]
fn error_value_catches_untested_ok_case() {
    // By default this tree should fail because it's configured to
    // generate an error value, and the tests forgot to check that
    // the code under test does return Ok.
    let tmp_src_dir = copy_of_testdata("error_value");
    run()
        .arg("mutants")
        .args(["-v", "-V", "--no-times", "--no-shuffle"])
        .arg("-d")
        .arg(tmp_src_dir.path())
        .assert()
        .code(2)
        .stderr("")
        .stdout(predicate::function(|stdout| {
            insta::assert_snapshot!(stdout);
            true
        }));
}
