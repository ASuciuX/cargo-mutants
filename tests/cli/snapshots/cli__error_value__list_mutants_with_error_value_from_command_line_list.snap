---
source: tests/cli/error_value.rs
expression: stdout
---
src/lib.rs:4:5: replace even_is_ok -> Result<u32, &'static str> with Ok(0)
src/lib.rs:4:5: replace even_is_ok -> Result<u32, &'static str> with Ok(1)
src/lib.rs:4:5: replace even_is_ok -> Result<u32, &'static str> with Err(::eyre::eyre!("mutant"))
src/lib.rs:4:14: replace == with != in even_is_ok

